holder.go

```diff
@@ -351,6 +359,42 @@ func (h *Holder) flushCaches() {
 	}
 }
 
+// setFileLimit attempts to set the open file limit to the FileLimit constant defined above.
+func (h *Holder) setFileLimit() {
+	lim := &syscall.Rlimit{}
+	var setLimit uint64
+	if err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, lim); err != nil {
+		h.logger().Printf("ERROR checking open file limit: %s", err)
+		return
+	}
+	if lim.Max < FileLimit {
+		h.logger().Printf("WARNING: open file limit max is %d. Please consider running \"ulimit -n %d\" before starting Pilosa to avoid \"too many open files\" error.", lim.Max, FileLimit)
+	}
+	// If the soft limit is lower than both the FileLimit constant and the hard limit, we will try to change it.
+	if lim.Cur < FileLimit && lim.Cur < lim.Max {
```

Comment:
Ok, I can give that a try. The hard limit is only mutable by root, but might as well at least try.